### PR TITLE
travis: backport 10.3 changes

### PR DESCRIPTION
**Pull request by grooverdan**
Also recommend enabling Travis auto-cancellation to prevent builds queuing up on the same branch if it isn't already. https://blog.travis-ci.com/2017-09-21-default-auto-cancellation.

Lets wait for this to finish. There's a substantial backlog of osx builds for the last day.

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -32,7 +32,7 @@ CXX := $(DEB_HOST_GNU_TYPE)-g++
 # at https://www.debian.org/doc/debian-policy/ch-source.html#s-debianrules-options
 ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
     NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
-    MAKEFLAGS += -j $(NUMJOBS)
+    MAKEFLAGS += -j$(NUMJOBS)
```
Comment (svoj) [inline debian/rules]: This was recently done by @ottok: 325c9ce9b39a7695bf217cd92fc36034b2e8207b

**Comment by svoj** [comment]
Taking into account recent activity around travis, I wonder who is to review this? @ottok @cvicentiu ?

File: debian/rules (hunk 1 above)
Comment (grooverdan) [reply debian/rules]: apparently this was just re-conforming to the debian policy docs with no material impacts.

**Comment by grooverdan** [comment]
Not quite following the milestones being a 10.2 target.

@cvicentiu commented ... https://github.com/MariaDB/server/pull/499#issuecomment-345993346

FYI - also testing https://travis-ci.org/grooverdan/mariadb-server/builds/309873867 based on few Travis bug comments.

**Comment by svoj** [comment]
Sorry, my bad.

**Comment by grooverdan** [comment]
Backported the 10.3 changes here. Added percona which is still a test in 10.2. Predictable failures on  rocksdb.mariadb_plugin and sys_vars.innodb_buffer_pool_load_at_startup_basic however all the other 10.3 fixes are included.

I added a few trivial patches at the end too.

**Comment by grooverdan** [comment]
https://github.com/MariaDB/server/pull/559#issuecomment-402417109 "Right now our biggest priority is to backport these Travis-CI improvements". Obvious not, since this still hasn't been merged. It doesn't resolve all 10.2 issues however it fix a lot of them.

**Comment by svoj** [comment]
@shinnok could you have a look at this if there's still anything to backport?

**Pull request closed without merging**